grpc/README.md

```diff
@@ -1,5 +1,225 @@
 # gRPC Service
 
+## gRPC Java Client 
+
+### Setup
+
+Let's see how to configure a java project to use the Stargate gRPC API.
+The generated code based on protobuf files (`query.proto` and `stargate.proto`) is shipped with the `grpc-proto` dependency.
+
+To see a guide how it compiles the Java code see: [gRPC setup project dependencies].
+In your client application, you only need to add one dependency:
+
+```xml
+<dependencies>
+    <dependency>
+        <groupId>io.stargate.grpc</groupId>
+        <artifactId>grpc-proto</artifactId>
+        <version>1.0.32</version>
+    </dependency>
+</dependencies>
+```
+The last missing piece is to add a functional channel service provider. We pick netty:
+```xml
+<dependencies>
+    <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-netty-shaded</artifactId>
+            <version>1.40.1</version>
+    </dependency>
+</dependencies>
+```
+If you do not add it, you will observe the following error:
+`No functional channel service provider found. Try adding a dependency on the grpc-okhttp, grpc-netty, or grpc-netty-shaded artifact`.
+
+Once we have all needed dependencies we should be able to use the Stargate gRPC-stub API. 
+After this step, you should have `StargateGrpc` available on your class path (from the `grpc-proto` dependency).
+It contains the logic for interacting with Stargate gRPC API.
+
+### Usage
+
+Once we have the generated code, we are ready to create the client based on that.
+Before delving into the code, we need to generate the auth token that will be used to perform Authorization, 
+to see how to do it, please visit the [Stargate Authz documentation].
+For the development purpose, this should be good enough:
+```shell script
+curl -X POST localhost:8081/v1/auth/token/generate --header "Content-Type: application/json" --data '{"key":"cassandra","secret":"cassandra"}'
+```
+assuming that Stargate is running on the `localhost:8081`  
+
+Once we have that, we can connect to the gRPC API. Firstly, we need to generate the `Channel` that is used to perform connection:
+```java
+public ManagedChannel createChannel(String host, int port) {
+    return ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
```

Comment:
maybe add a small note that `usePlaintext()` should only be used for testing and non-sensitive data? Actually removing it could be even better?

Reply:
@mpenick what's the recommended way for clients to create this channel?
When I drop `usePlaintext()`, I got:
```
Exception in thread "main" io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
Channel Pipeline: [SslHandler#0, ProtocolNegotiators$ClientTlsHandler#0, WriteBufferingAndExceptionHandler#0, DefaultChannelPipeline$TailContext#0]
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.stargate.proto.StargateGrpc$StargateBlockingStub.executeQuery(StargateGrpc.java:236)
	at GrpcClient.executeQuery(GrpcClient.java:27)
	at GrpcClient.main(GrpcClient.java:61)
```
locally

Reply:
Ivan is correct, `usePlaintext()` should only be used for development and testing. When using in a production environment it should use a load balancer that terminates TLS or the gRPC needs to support TLS (which it currently doesn't).

Reply:
Do you have an example from the client perspective how this should be configured?

Reply:
I mentioned it on:
https://github.com/stargate/stargate/pull/1236/commits/665ce94e4bb88f0d415fd113c93a9e491fbfc9f2